src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2156,8 +2166,10 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             CompilationTestData testData,
             CancellationToken cancellationToken)
         {
-            bool embedPdb = options?.DebugInformationFormat == DebugInformationFormat.Embedded;
+            options = options ?? EmitOptions.Default.WithIncludePrivateMembers(metadataPEStream == null);
+            bool embedPdb = options.DebugInformationFormat == DebugInformationFormat.Embedded;
             Debug.Assert(!embedPdb || pdbStream == null);
+            Debug.Assert(metadataPEStream == null || !options.IncludePrivateMembers);
```

Comment:
Logically, "A implies B" means "not A or B". So if "no secondary stream" implies "include private members", the assertion should be "there is a secondary stream OR include private members" which would be written
``` c#
    Debug.Assert(metadataPEStream != null || options.IncludePrivateMembers);
```
But that is not what is written.

I remain confused.